pontoon/administration/management/commands/sync_projects.py

```diff
@@ -0,0 +1,363 @@
+        db_project.last_synced = timezone.now()
+        db_project.save()
+
+        self.log('Synced project {0}', db_project.slug)
+
+    def handle_entity(self, changeset, db_project, key, db_entity, vcs_entity):
+        """
+        Determine what needs to be synced between the database and VCS versions
+        of a single entity and log what needs to be changed in the changeset.
+        """
+        if vcs_entity is None:
+            if db_entity is None:
+                # This should never happen. What? Hard abort.
+                raise CommandError('No entities found for key {0}'.format(key))
+            else:
+                # VCS no longer has the entity, remove it from Pontoon.
+                changeset.obsolete_db_entity(db_entity)
+        elif db_entity is None:
+            # New VCS entities are added to Pontoon.
+            changeset.create_db_entity(vcs_entity)
+        else:
+            for locale in db_project.locales.all():
+                if db_entity.has_changed(locale.code):
+                    # Pontoon changes overwrite whatever VCS has.
+                    changeset.update_vcs_entity(locale.code, db_entity, vcs_entity)
+                elif vcs_entity.has_translation_for(locale.code):
+                    # If Pontoon has nothing or has not changed, and the VCS
+                    # still has the entity, update Pontoon with whatever may
+                    # have changed.
+                    changeset.update_db_entity(locale.code, db_entity, vcs_entity)
+
+    def update_resources(self, db_project, vcs_project):
+        """Update the database on what resource files exist in VCS."""
+        relative_paths = vcs_project.resources.keys()
+        db_project.resource_set.exclude(path__in=relative_paths).delete()
+
+        for relative_path, vcs_resource in vcs_project.resources.items():
+            resource, created = db_project.resource_set.get_or_create(path=relative_path)
+            resource.format = Resource.get_path_format(relative_path)
+            resource.entity_count = len(vcs_resource.entities)
+            resource.save()
+
+    def get_vcs_entities(self, vcs_project):
+        return {self.entity_key(entity): entity for entity in vcs_project.entities}
+
+    def get_db_entities(self, db_project):
+        entities = (Entity.objects
+                    .select_related('resource')
+                    .filter(resource__project=db_project, obsolete=False))
+        return {self.entity_key(entity): entity for entity in entities}
+
+    def entity_key(self, entity):
+        """
+        Generate a key for the given entity that is unique within the
+        project.
+        """
+        return ':'.join([entity.resource.path, entity.string])
+
+    def commit_changes(self, db_project):
+        """Commit the changes we've made back to the VCS."""
+        # TODO: Set translation author properly.
+        user = User.objects.filter(is_superuser=True).first()
```

Comment:
Giving credit to the recent most translator is good enough, because there aren't many occasions when two localizers submit translations for the same project since last sync.

If there's more than one translator, listing the rest of them in the commit message would be better.

And the best we can do is probably this:
https://bugzilla.mozilla.org/show_bug.cgi?id=1179213